app/src/main/java/app/revanced/manager/ui/viewmodel/MainViewModel.kt

```diff
@@ -19,12 +35,38 @@ class MainViewModel(
     private val patchBundleRepository: PatchBundleRepository,
     private val patchSelectionRepository: PatchSelectionRepository,
     private val keystoreManager: KeystoreManager,
+    private val reVancedAPI: ReVancedAPI,
+    private val app: Application,
     val prefs: PreferencesManager
 ) : ViewModel() {
+    var updatedManagerVersion: String? by mutableStateOf(null)
+        private set
+
+    init { viewModelScope.launch { checkForUpdates() } }
+
+    fun dismissUpdateDialog() { updatedManagerVersion = null }
+
+    private suspend fun checkForUpdates() {
+        try {
+            if (prefs.managerAutoUpdates.get()) {
+                reVancedAPI.getRelease("revanced-manager").getOrThrow().metadata.tag.let {
+                    if(it != Build.VERSION.RELEASE) {
+                        updatedManagerVersion = it
+                    }
```

Comment:
`takeIf` can be used for this. You may have to give the parameter a name to avoid shadowing `it`
```suggestion
                   updatedManagerVersion = it.takeIf { it != Build.VERSION.RELEASE }
```